### DISCUSSION TITLE
What do you think of a :terminal mode?

### DISCUSSION THREAD

**Discussion by t-cadet** [helix-editor/helix#8580] [Ideas] [answered]
Vim & Emacs have a mode that lets the user interact with the terminal in a buffer of the editor.
The idea is to use the editor's usual movements & actions to manipulate the shell output or input commands efficiently, e.g. copying path, writing multi-line commands etc.

I think such a mode would be a great addition to Helix. 

For example it could be a mode where pressing enters executes a shell command.
In input and normal mode it would execute the current line as a shell command, 
in selection mode it would execute the current selection, 
and in both cases it would append the output on the next line.

What do you think? does it seem useful? easy/difficult to implement?

**Comment by the-mikedavis** [comment, marked as answer]
See https://github.com/helix-editor/helix/issues/1976
